application/core/LSMessageSource.php

```diff
@@ -0,0 +1,149 @@
+     * @see CDbMessageSource::cachingDuration
+     * @see CGettextMessageSource::cachingDuration
+     */
+    public $cachingDuration = 0;
+    /**
+     * @var string the ID of the cache application component that is used to cache the messages.
+     * Defaults to 'cache' which refers to the primary cache application component.
+     * @see CDbMessageSource::cacheID
+     * @see CGettextMessageSource::cacheID
+     */
+    public $cacheID = 'cache';
+    /**
+     * @var string the base path for all translated messages. Defaults to null, meaning
+     * @see CGettextMessageSource::basePath
+     */
+    public $basePath;
+    /**
+     * @var boolean whether to load messages from MO files. Defaults to true.
+     * @see CGettextMessageSource::useMoFile
+     */
+    public $useMoFile = true;
+    /**
+     * @var boolean whether to use Big Endian to read MO files.
+     * Defaults to false. This property is only used when {@link useMoFile} is true.
+     * see CGettextMessageSource::useBigEndian
+     */
+    public $useBigEndian = false;
+    /**
+     * @var string the message catalog name. This is the name of the message file (without extension)
+     * that stores the translated messages. Defaults to 'messages'.
+     * see CGettextMessageSource::catalog
+     */
+    public $catalog = 'messages';
+
+    /**
+     * Loads the message translation for the specified language and category.
+     * @see CGettextMessageSource::loadMessages
+     * @see CDbMessageSource::loadMessages
+     * @param string $category the message category
+     * @param string $language the target language
+     * @return array the loaded messages
+     */
+    protected function loadMessages($category, $language)
+    {
+        $messageFile = $this->basePath . DIRECTORY_SEPARATOR . $language . DIRECTORY_SEPARATOR . $this->catalog;
+        if ($this->useMoFile) {
+            $messageFile .= self::MO_FILE_EXT;
+        } else {
+            $messageFile .= self::PO_FILE_EXT;
+        }
+
+        if (
+            $this->cachingDuration > 0
+            && $this->cacheID !== false
+            && ($cache = Yii::app()->getComponent($this->cacheID)) !== null
+        ) {
+            $key = self::CACHE_KEY_PREFIX . $messageFile . $category . '.' . $language;
+            if (($data = $cache->get($key)) !== false) {
+                return unserialize($data);
+            }
+        }
+
+        /* Messages by getext */
+        if (is_file($messageFile)) {
+            if ($this->useMoFile) {
+                $file = new CGettextMoFile($this->useBigEndian);
+            } else {
+                $file = new CGettextPoFile();
+            }
+            $messagesGettext = $file->load($messageFile, $category);
+        } else {
+            $messagesGettext = array();
+        }
+
+        /* Messages by DB */
+        if (App()->getConfig('DBVersion') >= 480) {
+            $messagesDb = $this->loadMessagesFromDb($category, $language);
```

Comment:
Wait, does this mean the there's a db hit for every translation string...?

Reply:
Or this only happens once, at app start?

Reply:
Happens once at each call if debug>0, like reading po/mo

Else : file caching by default.

Reply:
:star: